File: backend/helpers/validation.py
import re

def validate_email(email: str) -> bool:
    pattern = r'^[^\s@]+@[^\s@]+\.[^\s@]+$'
    return bool(re.match(pattern, email))

def validate_account_number(account_number: str) -> bool:
    if len(account_number) != 10:
        return False
    if not account_number.startswith("180"):
        return False
    if not account_number.isdigit():
        return False

    base = account_number[:-1]
    check_digit = sum(int(d) for d in base) % 10

    return int(account_number[-1]) == check_digit

File: backend/src/helpers/validation.js
function generateAccountNumber(userId) {
  const base = '180' + String(userId).padStart(6, '0');
  const checkDigit = [...base].reduce((s, d) => s + Number(d), 0) % 10;
  return base + checkDigit;
}

function validateAccountNumber(num) {
  if (!/^\d{10}$/.test(num)) return false;
  if (!num.startsWith('180')) return false;
  const base = num.slice(0, -1);
  const check = Number(num.slice(-1));
  const sum = [...base].reduce((s, d) => s + Number(d), 0) % 10;
  return sum === check;
}

module.exports = { generateAccountNumber, validateAccountNumber };
